### ISSUE TITLE
option band '60g' does not properly set hwmode to 'ad' in generated hostapd-phyX.conf on WiGig devices

### ISSUE THREAD

**Issue by phansel** [openwrt/openwrt#11771]
### Describe the bug

From https://openwrt.org/docs/guide-user/network/wifi/basic#wi-fi_interfaces:

Name | Type | Required | Default | Description
-- | -- | -- | -- | --
hwmode | string | no | (driver default) | Specifies the hardware mode, possible values are 11b for 2.4 GHz (used for legacy 802.11b only), 11g for 2.4 GHz (used for 802.11b, 802.11g and 802.11n) and 11a for 5 GHz  (used for 802.11a, 802.11n and 802.11ac). Note that 11ng, 11na, 11n and  11ac are invalid options, this setting should largely be seen as  controlling the frequency band. **(DEPRECATED since 21.02.2, replaced with band)**
band | string | no | (driver default) | Specifies the band, possible values are 2g for 2.4 GHz, 5g for 5 GHz, 6g for 6 GHz and **60g for 60 GHz**. (NEW since 21.02.2, replaces hwmode)

The 'band' wifi-device option does not function as expected when the radio is configured as an AP on WiGig devices (at least, the Talon AD7200). It passes the wrong 'hwmode' parameter to hostapd ('a' rather than 'ad'). 

`option band '60g'` does work for devices in station mode.

### OpenWrt version

r16688-fa9a932fdb

### OpenWrt target/subtarget

ipq806x/generic

### Device

TP-Link Talon AD7200

### Image kind

Self-built image

### Steps to reproduce

Create an access point on 60GHz via radio0 and default_radio0 (named wlan0) with `option band '60g'` set under wifi-device 'radio0'.  Check the interface status a few moments later by running `iw dev wlan0 info` and check the generated hostapd config by reading /var/run/hostapd-phy0.conf.

### Actual behaviour

/var/run/hostapd-phy0.conf will contain the line hw_mode=a.

hostapd silently fails to activate the AP, leaving the device as 'type managed'. 

### Expected behaviour

/var/run/hostapd-phy0.conf will contain the line hw_mode=ad.

hostapd should successfully launch the AP with an ssid and with `type AP`.

### Additional info

The AP successfully launches if the two following conditions are met:

- `option hwmode 'ad'` must be set. 
- `option band 'ad'` is set, or `option band` is not set at all.

### Diffconfig

```text
CONFIG_TARGET_ipq806x=y
CONFIG_TARGET_ipq806x_generic=y
CONFIG_TARGET_ipq806x_generic_DEVICE_tplink_ad7200=y
CONFIG_DROPBEAR_ECC=y
CONFIG_DROPBEAR_ECC_FULL=y
CONFIG_PACKAGE_firewall=m
CONFIG_PACKAGE_ip-bridge=y
CONFIG_PACKAGE_ip-full=y
CONFIG_PACKAGE_jansson=y
CONFIG_PACKAGE_kmod-bonding=y
CONFIG_PACKAGE_kmod-crypto-aead=y
CONFIG_PACKAGE_kmod-crypto-ccm=y
CONFIG_PACKAGE_kmod-crypto-cmac=y
CONFIG_PACKAGE_kmod-crypto-crc32c=y
CONFIG_PACKAGE_kmod-crypto-ctr=y
CONFIG_PACKAGE_kmod-crypto-gcm=y
CONFIG_PACKAGE_kmod-crypto-gf128=y
CONFIG_PACKAGE_kmod-crypto-ghash=y
CONFIG_PACKAGE_kmod-crypto-hash=y
CONFIG_PACKAGE_kmod-crypto-hmac=y
CONFIG_PACKAGE_kmod-crypto-manager=y
CONFIG_PACKAGE_kmod-crypto-null=y
CONFIG_PACKAGE_kmod-crypto-pcompress=y
CONFIG_PACKAGE_kmod-crypto-rng=y
CONFIG_PACKAGE_kmod-crypto-seqiv=y
CONFIG_PACKAGE_kmod-crypto-sha256=y
CONFIG_PACKAGE_kmod-gre=y
CONFIG_PACKAGE_kmod-gre6=y
CONFIG_PACKAGE_kmod-ip6-tunnel=y
CONFIG_PACKAGE_kmod-iptunnel=y
CONFIG_PACKAGE_kmod-iptunnel6=y
CONFIG_PACKAGE_kmod-lib-crc16=y
CONFIG_PACKAGE_kmod-lib-crc32c=y
CONFIG_PACKAGE_kmod-nfnetlink=y
CONFIG_PACKAGE_kmod-nft-core=y
CONFIG_PACKAGE_kmod-nft-fib=y
CONFIG_PACKAGE_kmod-nft-nat=y
CONFIG_PACKAGE_kmod-nft-offload=y
CONFIG_PACKAGE_kmod-trelay=m
CONFIG_PACKAGE_kmod-tun=y
CONFIG_PACKAGE_libbpf=y
CONFIG_PACKAGE_libelf=y
CONFIG_PACKAGE_libmnl=y
CONFIG_PACKAGE_libnftnl=y
CONFIG_PACKAGE_librt=y
CONFIG_PACKAGE_nftables-json=y
CONFIG_PACKAGE_procd-seccomp=y
CONFIG_PACKAGE_relayd=y
CONFIG_PACKAGE_resolveip=y
CONFIG_PACKAGE_zlib=y
CONFIG_PACKAGE_kmod-ipt-conntrack=y
CONFIG_PACKAGE_kmod-ipt-nat=y
```

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by dhewg** [comment]
> r16688-fa9a932fdb

That's ancient:
```
$ ./scripts/getver.sh r16688
f70e11cd974effdea1ed2ae0ded3033984ba3dad
$ git show f70e11cd974effdea1ed2ae0ded3033984ba3dad
...
Date:   Fri May 7 01:00:58 2021 +0200
...
```

It sounds like this patch https://git.openwrt.org/?p=project/netifd.git;a=commitdiff;h=2e1fcf4cd61222fc0d307d863da71d17fbd34d6f fixed it.
So just an update would probably resolve that issue.

**Comment by jow-** [comment]
As pointed out above, already fixed with 2e1fcf4cd61222fc0d307d863da71d17fbd34d6f
